source/slang/slang-ir-specialize-function-call.cpp

```diff
@@ -31,6 +31,9 @@ bool FunctionCallSpecializeCondition::isParamSuitableForSpecialization(IRParam*
         //
         if (as<IRGlobalParam>(arg)) return true;
 
+        // Similarly for these global values
+        if( as<IRGlobalValueWithCode>(arg) ) return true;
```

Comment:
An `IRGlobalValueWithCode` can be a `IRGeneric`, `IRFunc`, or simply a constant with init expr.
Do you intend to include all of them here?

Reply:
Yes, although it's not technically necessary for the hof work.

We could even expand it to every global as they're all *suitable* for specialization. It could be that I've misinterpreted this function, but I thought it was more or less "if by specializing the function at this parameter, would it escape its scope?", i.e. is it globally scoped